addon/utils/serialize-and-push.js

```diff
@@ -0,0 +1,10 @@
+import { isArray } from '@ember/array';
+
+export default function(options, response) {
+  let modelName = this.constructor.modelName || this.constructor.typeKey;
+  let modelClass = this.store.modelFor(modelName);
+  let serializer = this.store.serializerFor(modelName);
```

Comment:
I've in `build-url.js`, I've broken out methods for getting model name and class. Please update to use those, so that we don't have some of the code using `model.constructor` and other places using `store.modelFor`

Reply:
Thanks, I've changed `serializeAndPush` to use these.